src/graphql/data/nft/Details.ts

```diff
@@ -93,92 +91,85 @@ const detailsQuery = graphql`
   }
 `
 
-export function useLoadDetailsQuery(address?: string, tokenId?: string): void {
-  const [, loadQuery] = useQueryLoader(detailsQuery)
-  useEffect(() => {
-    if (address && tokenId) {
-      loadQuery({ address, tokenId })
-    }
-  }, [address, tokenId, loadQuery])
-}
-
-export function useDetailsQuery(address: string, tokenId: string): [GenieAsset, CollectionInfoForAsset] | undefined {
-  const queryData = useLazyLoadQuery<DetailsQuery>(
-    detailsQuery,
-    {
+export function useNftAssetDetails(
+  address: string,
+  tokenId: string
+): { data: [GenieAsset, CollectionInfoForAsset]; loading: boolean } {
+  const { data: queryData, loading } = useDetailsQuery({
+    variables: {
       address,
       tokenId,
     },
-    { fetchPolicy: 'store-or-network' }
-  )
+    fetchPolicy: 'cache-and-network',
+  })
 
-  const asset = queryData.nftAssets?.edges[0]?.node
+  const asset = queryData?.nftAssets?.edges[0]?.node as NonNullable<NftAsset>
```

Comment:
```suggestion
  const asset = queryData?.nftAssets?.edges[0]?.node as NonNullable<NftAsset> | undefined
```

isn't that the actual type?